src/components/Layout/DefaultLayout/DefaultAxis.js

```diff
@@ -56,10 +56,10 @@ const DefaultAxis = ({ axisId, className }) => {
         const dimensions = dimensionIds.map((id) => {
             let dimension = {}
             if (metadata[id]) {
-                dimension = metadata[id]
+                dimension = { ...metadata[id] }
```

Comment:
Is there any possibility that the metadata of the dimensions could have non-simple property types (object or array) that would therefore retain referential equality? I wasn't able to find any examples of that, so it looks safe. But just wanted to check with you on that.